### PR TITLE
Feature/asv 1529 wallet install offer notification

### PR DESCRIPTION
**Pull request by paquiquinho**
**What does this PR do?**

This PR aims to implement a new notification to be sent 5 minutes after an app with wallet promotion reaches the claimable state. 

**Database changed?**

   Yes - RealmLocalNotificationSync added

**Where should the reviewer start?**

- [x] NotificationCenter.java
- [x] SystemNotificationShower.java

**How should this be manually tested?**

Change the time to something more reasonable for testing and download an app with promotion+wallet. This should send a notification after both installs are done. Note that claiming the promotion should cancel a scheduled notification that hasn't yet showed;

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1529](https://aptoide.atlassian.net/browse/ASV-1529)

**Code Review Checklist**

- [x] Architecture
- [x] Documentation on public interfaces
- [x] Database changed?
- [x] If yes - Database migration?
- [x] Remove comments & unused code & forgotten testing Logs
- [x] Codestyle
- [x] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/sync/alarm/SyncStorage.java [hunk 1]
```diff
@@ -8,24 +10,35 @@
...
 
   public Sync get(String syncId) {
-    return syncs.get(syncId);
+    if (syncId.equals(LocalNotificationSync.APPC_CAMPAIGN_NOTIFICATION)) {
+      return persistence.get()
+          .toBlocking()
+          .first();
+    } else {
+      return syncs.get(syncId);
+    }
```
Comment (jdandrade) [inline app/src/main/java/cm/aptoide/pt/sync/alarm/SyncStorage.java]: I can't remember, did we end up not finding a solution to refactor this to be an observable - syncs.get()

File: app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java [hunk 2]
```diff
@@ -902,6 +904,18 @@ private void handleClickOnRetry() {
...
+                && walletPromotionViewModel.isAppViewAppInstalled()) {
+              appViewManager.scheduleNotification(
+                  String.valueOf(walletPromotionViewModel.getAppcValue()),
+                  appViewViewModel.getIcon(), "aptoideinstall://package="
+                      + appViewViewModel.getPackageName()
+                      + "&store="
+                      + appViewViewModel.getStore()
+                      .getName()
+                      + "&show_install_popup=false");
+            }
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java]: fixed (notification built in manager)

File: app/src/main/java/cm/aptoide/pt/notification/AppcPromotionNotificationStringProvider.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import android.content.Context;
+import cm.aptoide.pt.R;
+
+public class AppcPromotionNotificationStringProvider {
+
+  private final Context context;
+
+  public AppcPromotionNotificationStringProvider(Context context) {
+    this.context = context;
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/notification/AppcPromotionNotificationStringProvider.java]: fixed

File: app/src/main/java/cm/aptoide/pt/notification/NotificationReceiver.java [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
 
+import android.app.NotificationManager;
 import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
 import android.os.Bundle;
 import cm.aptoide.pt.AptoideApplication;
 import com.jakewharton.rxrelay.PublishRelay;
 
+import static android.content.Context.NOTIFICATION_SERVICE;
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/notification/NotificationReceiver.java]: fixed

File: app/src/main/java/cm/aptoide/pt/sync/alarm/SyncStorage.java (hunk 1 above)
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/sync/alarm/SyncStorage.java]: we came up with no alternative solution for this, hence the toBlocking here;

**Pull request merged**